context.c

```diff
@@ -62,7 +64,7 @@ void ctx_free(struct Context **ptr)
   mutt_hash_free(&ctx->thread_hash);
   notify_free(&ctx->notify);
 
-  FREE(ptr);
+  STAILQ_REMOVE(&ContextList, ctx, Context, entries);
```

Comment:
you still need to free the `ctx` yourself, `STAILQ_REMOVE` only deals with unlinking from the list

Reply:
whoops, nice catch.